compiler/x/env/OMRCPU.cpp

```diff
@@ -28,6 +28,30 @@
 #include "x/runtime/X86Runtime.hpp"
 
 
+OMR::X86::CPU::CPU() :
+   OMR::CPU()
+   {
+   // Only enable the features that compiler currently uses
+   uint32_t enabledFeatures [] = {OMR_FEATURE_X86_FPU, OMR_FEATURE_X86_CX8, OMR_FEATURE_X86_CMOV,
+                                  OMR_FEATURE_X86_MMX, OMR_FEATURE_X86_SSE, OMR_FEATURE_X86_SSE2,
+                                  OMR_FEATURE_X86_SSSE3, OMR_FEATURE_X86_SSE4_1, OMR_FEATURE_X86_POPCNT,
+                                  OMR_FEATURE_X86_AESNI, OMR_FEATURE_X86_OSXSAVE, OMR_FEATURE_X86_AVX,
+                                  OMR_FEATURE_X86_HLE, OMR_FEATURE_X86_RTM};
+
+   memset(_featureMasks, 0, OMRPORT_SYSINFO_FEATURES_SIZE*sizeof(uint32_t));
+   for (size_t i = 0; i < sizeof(enabledFeatures)/sizeof(uint32_t); i++)
+      {
+      uint32_t index = enabledFeatures[i]/32;
+      uint32_t shift = enabledFeatures[i]%32;
+      _featureMasks[index] |= (1 << shift);
```

Comment:
This bit shift relies on undefined behaviour. The literal constant `1` will have type `int` and shifting it by 31 bits would lead to signed overflow, which is undefined behaviour. This also assumes `int` is 32 bits, which is not guaranteed, so using `static_cast<uint32_t>(1)` is safer than simply using `1` (using `UINT32_C(1)` would also work if it was supported by all our build compilers, but I don't know if it is).
```suggestion
      _featureMasks[index] |= (static_cast<uint32_t>(1) << shift);
```

Reply:
@Leonardo2718 Is it also undefined behaviour for c then? Port library is doing something similar:
```c
if ((NULL != desc) && (feature < (OMRPORT_SYSINFO_FEATURES_SIZE * 32))) {
		uint32_t featureIndex = feature / 32;
		uint32_t featureShift = feature % 32;

		rc = OMR_ARE_ALL_BITS_SET(desc->features[featureIndex], 1 << featureShift);
	}
```

Reply:
so I guess `1u` is needed..

Reply:
Yes, it is also undefined behaviour in c.

`1u` can work but I personally prefer a cast because it makes the size of the type explicit. Technically, `unsigned int` is not required to be 32-bits, although it should be in all our build compilers.